Reintroduce slow down when landing manually

MaEtUgR opened the pull request:
**Test data / coverage**
Implementation is in our public release and tested a lot. The port is untested (not even SITL).

**Describe problem solved by the proposed pull request**
Same as #11487. I'm **porting our solution here for comparison**. Discussing with @bresch about how to proceed with the altitude slowdown/smoothing.

**Additional context**
#10659
There was a global position controller wide slow down constraint when flying downwards and getting close to the ground before the flight task architecture. For auto flight it was added again but not for manual flight yet.

bresch commented on src/lib/FlightTasks/tasks/ManualAltitude/FlightTaskManualAltitude.cpp:
```diff
@@ -252,6 +265,27 @@ void FlightTaskManualAltitude::_respectMaxAltitude()
+
+	} else if (_sub_home_position->get().valid_alt) {
+		dist_to_ground = -(_position(2) - _sub_home_position->get().z);
+	}
```
This is really similar to https://github.com/PX4/Firmware/blob/f229444722132a2e71d92d425597faaadcb480c4/src/lib/FlightTasks/tasks/AutoMapper/FlightTaskAutoMapper.cpp#L149-L162
Which one is best? Should we use the same code for both?

bresch commented:
I tested this branch with the SmoothVel FlightTask and worked as expected:
https://logs.px4.io/plot_app?log=27b74a04-05a6-4489-936c-48ce2c2f4acc
![slow_landing](https://user-images.githubusercontent.com/14822839/53416608-264de580-39d4-11e9-9479-d98034cfec36.png)

@MaEtUgR The thing I might still want is to limit the upward velocity the same way.

MaEtUgR commented:
Devcall: @bresch tested and it seems viable for him. I need to add the slow, smooth takeoff the same way and then we can use this solution as first step. Improved constraint handling which was the idea in https://github.com/PX4/Firmware/pull/11487 can be added on top.

MaEtUgR commented:
I already ported the same thing to upstream some time ago 😄 : #11080

MaEtUgR commented:
@bresch I rebased on master and added the slow start upwards when close to ground part. I did a quick successful SITL test. Could you do your checks as well?

MaEtUgR commented:
![slowdown](https://user-images.githubusercontent.com/4668506/53678648-7e316880-3cc2-11e9-8079-8c2d114406b1.PNG)

bresch commented:
@MaEtUgR Cool, I'll try that tomorrow if possible.

bresch commented:
SITL result with Smooth position control (Velocity) sub mode:
![01_plot](https://user-images.githubusercontent.com/14822839/54019462-8dfdf080-418b-11e9-8736-0c2405a1b267.png)

bresch commented:
@PX4/testflights  Can you test that on a quadcopter please?
We need to test both `Smooth position control (velocity)` and `Smooth position control` of `MPC_POS_MODE`.
Please set `MPC_ACC_DOWN_MAX` and `MPC_ACC_UP_MAX` to 3 m/s^2 for nicer results

Thanks!

jorge789 commented:
Parameters tested on Pixracer v4: Smooth position control of MPC_POS_MODE; 3m/s/s for MPC_ACC_DOWN_MAX and 3m/s/s for MPC_ACC_UP_MAX, no issues were noted. Good flight in general.

https://review.px4.io/plot_app?log=bc7f38cc-da4b-46ad-bc7d-c33837659a90

Parameters tested on Pixhawk Pro v4Pro: Smooth position control of MPC_POS_MODE; 3m/s/s for MPC_ACC_DOWN_MAX and 3m/s/s for MPC_ACC_UP_MAX, no issues were noted. Good flight in general.

https://review.px4.io/plot_app?log=bc7f38cc-da4b-46ad-bc7d-c33837659a90

Parameters tested on Pixhawk Pro v4Pro: Smooth position control  (velocity) of MPC_POS_MODE; 3m/s/s for MPC_ACC_DOWN_MAX and 3m/s/s for MPC_ACC_UP_MAX, no issues were noted. Good flight in general.

https://review.px4.io/plot_app?log=2aff3e8a-cdcc-4075-8445-39b217efc403

LorenzMeier commented:
Thanks for testing!

Junkim3DR commented:
Tested on Pixhawk 4 v5, parameter tested were:
- Smooth position control for MPC_POS_MODE.
- 3m/s/s for MPC_ACC_DOWN_MAX.
- 3m/s/s for MPC_ACC_UP_MAX.

Issues Noted: 
- Parameters couldn't be changed while PR was loaded (Lost connection a few seconds after changing parameters). Solution: Load Master, change parameters and then load PR. 
- Not disarming automatically nor manually, kill switch not working.

https://review.px4.io/plot_app?log=43aece45-c1ea-49dd-a22d-5ff0d577bb6a

MaEtUgR commented:
One moment. This pr does not change anything with the parameters you listed. Ar you completely sure that it was the cause of your problems? Were there any follow-ups I missed?
